### PR TITLE
fix: Authorization for Action CUD

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
Puts authorizations in check at the server-side for an action's creation/update/delete.

Note: This branch is on top of https://github.com/formbricks/formbricks/pull/741, so should be merged after it!

## How to Test
Try hardcoding a different environmnetId or an actionId that you do not have access to! You should get an authzn error. Fundamental priciple followed is to take minimal input data for user and get most of it from the DB.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [ ] Added a screen recording or screenshots to this PR
- [ ] Filled out the "How to test" section in this PR
- [x] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/lib/actionClass/auth.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { hasUserEnvironmentAccess } from "../environment/auth";
+import { getActionClass } from "./service";
+import { unstable_cache } from "next/cache";
+
+export const canUserAccessActionClass = async (userId: string, actionClassId: string): Promise<boolean> =>
```
Comment (mattinannt) [inline packages/lib/actionClass/auth.ts:5]: I think it makes sense to validate inputs here, too (like in services)

File: apps/web/app/(app)/environments/[environmentId]/(actionsAndAttributes)/actions/actions.ts [hunk 2]
```diff
@@ -1,31 +1,116 @@
...
-  const response = {
-    activeSurveys: surveys.filter((s) => s.status === "inProgress").map((survey) => survey.name),
-    inactiveSurveys: surveys.filter((s) => s.status !== "inProgress").map((survey) => survey.name),
-  };
-  return response;
+  const session = await getServerSession(authOptions);
+  if (!session) throw new AuthorizationError("Not authorized");
+
+  const isAuthorized = await canUserAccessActionClass(session.user.id, actionClassId);
+  if (isAuthorized) {
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/(actionsAndAttributes)/actions/actions.ts:106]: I would prefer this coding style:

```ts
  const session = await getServerSession(authOptions);
  if (!session) throw new AuthorizationError("Not authorized");

  const isAuthorized = await canUserAccessActionClass(session.user.id, actionClassId);
 
  if (!isAuthorized) {
    throw new AuthorizationError("Not authorized");
  }

  const surveys = await getSurveysByActionClassId(actionClassId);
  const response = {
    activeSurveys: surveys.filter((s) => s.status === "inProgress").map((survey) => survey.name),
    inactiveSurveys: surveys.filter((s) => s.status !== "inProgress").map((survey) => survey.name),
  };
  return response;
```

Comment (mattinannt) [reply apps/web/app/(app)/environments/[environmentId]/(actionsAndAttributes)/actions/actions.ts:106]: same for all other actions

File: packages/lib/actionClass/auth.ts (hunk 1 above)
Comment (ShubhamPalriwala) [reply packages/lib/actionClass/auth.ts:5]: done

File: apps/web/app/(app)/environments/[environmentId]/(actionsAndAttributes)/actions/actions.ts (hunk 2 above)
Comment (ShubhamPalriwala) [reply apps/web/app/(app)/environments/[environmentId]/(actionsAndAttributes)/actions/actions.ts:106]: okay! done

**Review by mattinannt — APPROVED** [review]

**Pull request merged**